app/src/main/java/org/schabi/newpipe/util/image/ImageStrategy.java

```diff
@@ -0,0 +1,195 @@
+package org.schabi.newpipe.util.image;
+
+import static org.schabi.newpipe.extractor.Image.HEIGHT_UNKNOWN;
+import static org.schabi.newpipe.extractor.Image.WIDTH_UNKNOWN;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.schabi.newpipe.extractor.Image;
+
+import java.util.Comparator;
+import java.util.List;
+
+public final class ImageStrategy {
+
+    // when preferredImageQuality is LOW or MEDIUM, images are sorted by how close their preferred
+    // image quality is to these values (H stands for "Height")
+    private static final int BEST_LOW_H = 75;
+    private static final int BEST_MEDIUM_H = 250;
```

Comment:
Well, we want this to be a frontend thing, not an extractor thing. We want to be able to change it from NewPipe if we decide to create a better strategy or whatever. Also, these constants represent something different than those in `ResolutionLevel`: the values in `ResolutionLevel` are max-thresholds for the various categories, while these values are used to choose the image whose height is closest to them.

Reply:
makes sense.

Reply:
You might need to adjust the values a little, because when setting the image quality to high I get blurry thumbnails for YouTube but perfectly sharp thumbnails for the other services:
<details><summary>Screenshots</summary>

 ![Screenshot_20230921-163515_NewPipe multiple-images](https://github.com/TeamNewPipe/NewPipe/assets/17365767/a2bd50f2-6884-4f52-a8d0-3540e20039b8) ![Screenshot_20230921-163601_NewPipe multiple-images](https://github.com/TeamNewPipe/NewPipe/assets/17365767/08198b3b-1a98-4e94-8101-09cd7ca3339c) ![Screenshot_20230921-163607_NewPipe multiple-images](https://github.com/TeamNewPipe/NewPipe/assets/17365767/59b07908-2957-4fe4-b621-0e7aec65280a) ![Screenshot_20230921-163622_NewPipe multiple-images](https://github.com/TeamNewPipe/NewPipe/assets/17365767/3ff12948-0313-4844-8b76-28763f089b39) 

</details>

Reply:
YouTube doesn't provide better thumbnails for info items unfortunately. The best thumbnail is already being chosen. If you load video details you will see a better picture is loaded for YouTube, too.

Reply:
ah ok. In this case we'll need to do some changes in the extractor. I just checked the thumbnail URLs. They look like this:

Stream URL: https://www.youtube.com/watch?v=M9k_2aX7Z24
https://i.ytimg.com/vi/M9k_2aX7Z24/hqdefault.jpg?sqp=-oaymwEbCMQBEG5IVfKriqkDDggBFQAAiEIYAXABwAEG&rs=AOn4CLDkVXxfTq7DmG9_QfZ9cZtHa33QNQ
https://i.ytimg.com/vi/M9k_2aX7Z24/hqdefault.jpg?sqp=-oaymwEcCPYBEIoBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLBIEhn3OK2WJ02HqBfzaYxw_2KrYw
https://i.ytimg.com/vi/M9k_2aX7Z24/hqdefault.jpg?sqp=-oaymwEcCNACELwBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLCj_e440zgW4wR6f5qyZl8KikrUZA
https://i.ytimg.com/vi/M9k_2aX7Z24/maxresdefault.jpg

Stream URL: https://www.youtube.com/watch?v=b3unvqHR9m8
https://i.ytimg.com/vi/b3unvqHR9m8/hqdefault.jpg?sqp=-oaymwEcCPYBEIoBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLCci958CBvtcsnyOoXY23t-Jngdeg
https://i.ytimg.com/vi/b3unvqHR9m8/hqdefault.jpg?sqp=-oaymwEcCNACELwBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLC0KAPC8whJ6NYDrgFURSLezjLHQQ
https://i.ytimg.com/vi_webp/b3unvqHR9m8/maxresdefault.webp?v=650ab64d

Although the last URL is different from the maxres URL from above, this one also works, but is not provided by YouTube:
https://i.ytimg.com/vi/b3unvqHR9m8/maxresdefault.jpg

We might be able to just add them in the extractor even if the highres version is not provided. The URL seems to be
`https://i.ytimg.com/vi/<STREAM_ID>/maxresdefault.jpg`

CC @AudricV

Reply:
I think this can be taken care of separately simply with an update in the extractor

Reply:
As we discussed on our internal chat, such thumbnails are not provided by YouTube desktop client and `maxresdefault` variant is not available on some videos. This behavior will not change.